### ISSUE TITLE
euler训练protobuf冲突：File already exists in database: graph_service.proto

### ISSUE THREAD

**Issue by maishuguang** [alibaba/euler#115]
报错如下：
[libprotobuf ERROR external/protobuf_archive/src/google/protobuf/descriptor_database.cc:58] File already exists in database: graph_service.proto
[libprotobuf FATAL external/protobuf_archive/src/google/protobuf/descriptor.cc:1401] CHECK failed: generated_database_->Add(encoded_file_descriptor, size): 
terminate called after throwing an instance of 'google::protobuf::FatalException'
怀疑是由于euler和tensorflow都是静态链接了libprotobuf。
请问有遇到类似问题的吗，或者讨论下解决思路，感谢🙏

**Comment by lixusign** [comment]
我们是在改造hvd过程中遇到的上述问题，主要想尝试使用ringAllreduce看看效果，这个是使用hvd的改造代码

`def run_train(model, flags_obj, master, is_chief):
  utils_context.training = True

  batch_size = flags_obj.batch_size // model.batch_size_ratio
  if flags_obj.model == 'line' or flags_obj.model == 'randomwalk':
    source = euler_ops.sample_node(
        count=batch_size, node_type=flags_obj.all_node_type)
  else:
    source = euler_ops.sample_node(
        count=batch_size, node_type=flags_obj.train_node_type)
  source.set_shape([batch_size])
  _, loss, metric_name, metric = model(source)

  hvd.init()
  config = tf.ConfigProto()
  tf.ConfigProto(log_device_placement=True,allow_soft_placement=True)
 
  optimizer_class = optimizers.get(flags_obj.optimizer)
  optimizer = optimizer_class(learning_rate=flags_obj.learning_rate * hvd.size())
  optimizer = hvd.DistributedOptimizer(optimizer)
  global_step = tf.train.get_or_create_global_step()
  train_op = optimizer.minimize(loss, global_step=global_step)

  hooks = []

  tensor_to_log = {'step': global_step, 'loss': loss, metric_name: metric}
  hooks.append(
      tf.train.LoggingTensorHook(
          tensor_to_log, every_n_iter=flags_obj.log_steps))

  num_steps = int((flags_obj.max_id + 1) // batch_size * flags_obj.num_epochs)
  hooks.append(tf.train.StopAtStepHook(last_step=num_steps))

  if len(flags_obj.worker_hosts) == 0 or flags_obj.task_index == 1:
    hooks.append(
        tf.train.ProfilerHook(save_secs=180, output_dir=flags_obj.model_dir))
  if len(flags_obj.worker_hosts):
    hooks.append(utils_hooks.SyncExitHook(len(flags_obj.worker_hosts)))
  if hasattr(model, 'make_session_run_hook'):
    hooks.append(model.make_session_run_hook())

  hooks.append(hvd.BroadcastGlobalVariablesHook(0))

  checkpoint_dir = flags_obj.model_dir if hvd.rank() == 0 else None

  with tf.train.MonitoredTrainingSession(
      checkpoint_dir=flags_obj.model_dir,
      config=config,
      hooks=hooks) as sess:
    while not sess.should_stop():
      sess.run(train_op)`

**Comment by lixusign** [comment]
在吗？请问如何处理这种冲突？

**Comment by lixusign** [comment]
在吗? centos7
